homeassistant/components/imap/coordinator.py

```diff
@@ -166,7 +165,13 @@ def _decode_payload(part: Message) -> str:
             Falls back to the raw content part if decoding fails.
             """
             try:
-                return str(part.get_payload(decode=True).decode(self._charset))
+                content_charset = part.get_content_charset()
+                decoded_payload: bytes = part.get_payload(decode=True)
+                return (
+                    decoded_payload.decode(content_charset)
+                    if content_charset is not None
+                    else str(decoded_payload)
```

Comment:
Isn't `decoded_payload` bytes? What will this return then?

Reply:
When the decode param is set `bytes` are return. When no decode is set, it seems to returns a string, not bytes.
To make sure we have a string presentation the wrapper is added as we do not know the encoding.
Instead we could `try` decoding `UTF-8` as a default if the `charset` is not set on the message part?

Reply:
> When no decode is set, it seems to returns a string, not bytes.

Then the type annotation is wrong.